src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -486,13 +496,26 @@ object AppDefinition extends GeneralPurposeCombinators {
     * errors for every deployment potentially unrelated to the deployed apps.
     */
   def validAppDefinition(
-    enabledFeatures: Set[String])(implicit pluginManager: PluginManager): Validator[AppDefinition] =
+    enabledFeatures: Set[String], roleEnforcement: RoleEnforcement)(implicit pluginManager: PluginManager): Validator[AppDefinition] =
     validator[AppDefinition] { app =>
       app.id is valid and PathId.absolutePathValidator and PathId.nonEmptyPath
       app.dependencies is every(PathId.pathIdValidator)
-
+      app is validWithRoleEnforcement(roleEnforcement)
     } and validBasicAppDefinition(enabledFeatures) and pluginValidators
 
+  private def validWithRoleEnforcement(roleEnforcement: RoleEnforcement): Validator[AppDefinition] = validator[AppDefinition] { app =>
```

Comment:
I wonder if we should not call this instead of `getRoleEnforcement` in `AppsResource`:
```
  private def validWithRoleEnforcement(enforceRole: Boolean, validRoles: Set[String]): Validator[AppDefinition] = validator[AppDefinition] { app =>
  ...
}
```

What do you think?